### PR TITLE
[Feature] Add Unit and Integration Tests for StoryClient and IPAsset

### PR DESCRIPTION
**Pull request by aandrewchung**
## Description
This PR introduces extensive testing for the StoryClient and IPAsset components of the Python SDK. Key changes include:
1. Added unit tests for StoryClient and IPAsset.
2. Refactored story_client.py to remove unnecessary code and streamline class definition.
3. Implemented integration tests for IP asset registration functionality.
4. Ensured consistent code structure and organization for better maintainability.

## Test Plan 
1. Run unit tests for StoryClient and IPAsset:
- Use pytest to execute tests in tests/unit/ directory.
- Verify the correctness of StoryClient initialization and IPAsset registration logic.

2. Run integration tests for IP asset registration:
- Use pytest to execute tests in tests/integration/ directory.
- Validate the interaction

3. Review and confirm the proper organization and refactoring of the story_client.py file.

## Related Issue
n/a

## Notes
- Ensure the environment variables (WALLET_PRIVATE_KEY and RPC_PROVIDER_URL) are set correctly in the .env file for the tests to run successfully.

### CODE REVIEW

**Comment by aandrewchung** [comment]
for next pr: Update readme, maybe create contributing.md and security.md, create COC, create docs / requirements.doc ?

File: scripts/.env-example [hunk 1]
```diff
@@ -0,0 +1 @@
+API_KEY="YOUR-API-KEY-HERE"
```
Comment (DonFungible) [inline scripts/.env-example:1]: duplicate to the one in the root dir?

File: tests/unit/resources/test_ip_asset.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from web3 import Web3
+from dotenv import load_dotenv
+
+# Ensure the src directory is in the Python path
+current_dir = os.path.dirname(__file__)
+src_path = os.path.abspath(os.path.join(current_dir, '..', '..', '..'))
+if src_path not in sys.path:
+    sys.path.append(src_path)
+
+# Now you can import from src
```
Comment (DonFungible) [inline tests/unit/resources/test_ip_asset.py:15]: Remove comments where the line below it are self explanatory

**Review by DonFungible — APPROVED** [review]
lgtm. Will want to consider adding the unit tests to GHA in a future PR

**Review by edisonz0718 — APPROVED** [review]

**Comment by aandrewchung** [comment]
Make clear (maybe in the documentation) about .env and that they just have to create an account with the web3 library and then pass that accoutn to story_client

**Pull request merged**